_resources/port1.0/group/python-1.0.tcl

```diff
@@ -285,6 +292,12 @@ default destroot.target {[python_get_defaults destroot_target]}
 default python.pep517   no
 default python.pep517_backend   setuptools
 
+default python.test_framework   pytest
+default test.run        no
```

Comment:
This is slightly redundant. You can use `tbool` to handle it being unset.

Reply:
thanks for the pointer, I didn't know about the existence of that proc! 

It took me a while also to figure out I needed to add `test.run` to the list of `global` parameters in the `python_add_dependencies` proc. My Tcl knowledge is rusty at best... sorry for my ignorance, do I understand it correctly that variables are only available within a proc if they are defined as global the `global`? Does that then work both ways; in that, if you set a variable within a proc (without returning) will that be available in the "main" Portfile only when it is listed as a `global` value?

Reply:
For a proc you define yourself, yes, you need to declare globals first in order to read them, and yes, setting variables not declared global will set them only in the local scope.

Procs defined via the MacPorts machinery like `pre-build` or whatever "cheat" by running `global {*}[info globals]` for you at the start.